HHS/OPRE-OPS
# Ops 1199/need by date validation

## What changed

adds group data validation to ensure its in the future

## Issue

#1199

## How to test

1. review agreement
2. try to add/edit budget line with needByDate in the past
3. ❌ see error

## Screenshots

<img width="507" alt="image" src="https://github.com/HHS/OPRE-OPS/assets/4629398/bef218b4-b93c-4aa5-adbe-e5b3edf3177b">
